cartoframes/data/dataset.py

```diff
@@ -578,7 +578,8 @@ def _rows(self, df, cols, with_lnglat, geom_col):
                     csv_row += '{val}|'.format(val=val)
 
             if the_geom_val is not None:
-                geom = decode_geometry(the_geom_val)
+                enc_type = detect_encoding_type(the_geom_val)
```

Comment:
We could assume every geometry has the same encoding, so I'd do this just once and save the enc_type as a variable ouf of the loop or as an instance attribute.

Reply:
:+1: